packages/@expo/cli/src/run/android/resolveGradlePropsAsync.ts

```diff
@@ -34,11 +42,35 @@ export function resolveGradleProps(
   // This won't work for non-standard flavor names like "myFlavor" would be treated as "my", "flavor".
   const [buildType, ...flavors] = variant.split(/(?=[A-Z])/).map((v) => v.toLowerCase());
   const apkVariantDirectory = path.join(apkDirectory, ...flavors, buildType);
+  const architectures = await getConnectedDeviceABIS(buildType, options.allArch, device);
 
   return {
     appName,
     buildType,
     flavors,
     apkVariantDirectory,
+    architectures,
   };
 }
+
+async function getConnectedDeviceABIS(
+  buildType: string,
+  allArch?: boolean,
+  device?: Device
+): Promise<string> {
+  if (allArch || buildType !== 'debug') {
+    return '';
+  }
+
+  if (device) {
+    const abis = await getDeviceABIsAsync(device);
+    return abis.join(',');
+  }
+
+  const devices = await getAttachedDevicesAsync();
+  const abisPromises = devices.map(async (device) => await getDeviceABIsAsync(device));
+  const abis = (await Promise.all(abisPromises))
+    .flat()
+    .filter((abi, i, arr) => arr.indexOf(abi) === i);
+  return abis.join(',');
+}
```

Comment:
I would make the `device` param required, `resolveDeviceAsync` should [always return an `AndroidDeviceManager` instance](https://github.com/expo/expo/blob/main/packages/%40expo/cli/src/run/android/resolveDevice.ts#L7-L23). And with that, we can simplify to:

```suggestion
async function getConnectedDeviceABIS(
  device: Device,
  buildType: string,
  allArch?: boolean
): Promise<string> {
  // Follow the same behavior as iOS, only enable this for debug builds
  if (allArch || buildType !== 'debug') {
    return '';
  }

  const abis = await getDeviceABIsAsync(device);
  return abis.join(',');
}
```